modules/systemdstates/systemdstates.go

```diff
@@ -0,0 +1,79 @@
+package systemdstates
+
+import (
+	"github.com/netdata/go-orchestrator/module"
+	"github.com/netdata/go.d.plugin/pkg/matcher"
+)
+
+type Config struct {
+	Selector matcher.SimpleExpr `yaml:"selector"`
+}
+
+func init() {
+	creator := module.Creator{
+		Defaults: module.Defaults{
+			Disabled:    false,
+			UpdateEvery: 1,
+		},
+		Create: func() module.Module { return New() },
+	}
+
+	module.Register("systemdstates", creator)
+}
+
+// New creates SystemdStates with default values
+func New() *SystemdStates {
+	return &SystemdStates{
+		charts:  charts.Copy(),
+		metrics: make(map[string]int64),
+	}
+}
+
+// SystemdStates SystemdStates module
+type SystemdStates struct {
+	module.Base  // should be embedded by every module
+	Config       `yaml:",inline"`
+	metrics      map[string]int64
+	charts       *module.Charts
+	unitsMatcher matcher.Matcher
+}
+
+// Cleanup makes cleanup
+func (SystemdStates) Cleanup() {}
+
+// Init makes initialization
+func (s *SystemdStates) Init() bool {
+
+	if !s.Selector.Empty() {
+		m, err := s.Selector.Parse()
+		if err != nil {
+			s.Errorf("error on creating per user stats matcher : %v", err)
+		}
+		s.unitsMatcher = matcher.WithCache(m)
+	}
+
+	return true
+}
+
+// Check makes check
+func (SystemdStates) Check() bool {
+	return true
+}
```

Comment:
I think this should be simple `return len(s.Collect()) > 0`.

We use check to ensure the module is able to collect data (plugin invokes Check() before starting the module).